corehq/apps/export/export.py

```diff
@@ -283,6 +292,29 @@ def get_export_writer(export_instances, temp_path, allow_pagination=True):
 def get_export_download(export_instances, filters, filename=None):
     from corehq.apps.export.tasks import populate_export_download_task
 
+    queue_len = get_queue_length(EXPORT_DOWNLOAD_QUEUE)
+    if queue_len == QUEUE_LENGTH_LOGGING_THRESHHOLD:
+        # This makes the assumption, which may not be true, that the
+        logger.info('Queue {} at 100 tasks. Logging task data.'.format(EXPORT_DOWNLOAD_QUEUE))
+        for task in get_queue_tasks(EXPORT_DOWNLOAD_QUEUE):
+            insts_str = '), ('.join((
+                '; '.join((
+                    'ID: {}'.format(inst['id']),
+                    'Name: {}'.format(inst['name']),
+                    'Domain: {}'.format(inst['domain']),
+                )) for inst in task['export_instances']
+            ))
+            task_str = '; '.join((
+                'Task ID: {}'.format(task['id']),
+                'State: {}'.format(task['state']),
+                'Status: {}'.format(task['status'].state),
+                'Start time: {}'.format(task['start_time'] or 'N/A'),
+                'Progress: {}%'.format(task['status'].progress.percent),
+                'Error: "{}"'.format(task['status'].progress.error_message or "N/A"),
+                'Export instances: [({})]'.format(insts_str),
+            ))
+            logger.info(task_str)
+
```

Comment:
Would you mind moving this into a helper function?  It's tangential to the main purpose of this method, and would be nice to keep it simple.